src/ansi-c/c_nondet_symbol_factory.h

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************\
+
+Module: C Nondet Symbol Factory
+
+Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#ifndef CPROVER_ANSI_C_C_NONDET_SYMBOL_FACTORY_H
+#define CPROVER_ANSI_C_C_NONDET_SYMBOL_FACTORY_H
+
+#include <util/message.h>
+#include <util/std_code.h>
+#include <util/symbol_table.h>
+
+class input_recordert
+{
+  struct input_datat
+  {
+    irep_idt base_name;
+    exprt expr;
+
+    input_datat(const irep_idt base_name, const exprt &expr):
+      base_name(base_name),
+      expr(expr)
+    {}
+  };
+  source_locationt location;
+  std::vector<struct input_datat> inputs;
+
+public:
+  explicit input_recordert(const source_locationt location):location(location)
+  {}
+  void record_input(const irep_idt &base_name, const exprt &expr);
+  void declare_recorded_inputs(code_blockt &init_code);
+};
+
+exprt c_nondet_symbol_factory(
+  const code_typet::parametert &p,
+  const int param_number,
+  const typet &type,
+  code_blockt &init_code,
+  bool allow_null,
+  symbol_tablet &symbol_table,
+  const source_locationt &,
+  message_handlert &message_handler,
+  input_recordert &input_recorder);
+
+exprt c_get_nondet_bool(const typet &);
+
+symbolt &c_new_tmp_symbol(
+  symbol_tablet &symbol_table,
+  const source_locationt &loc,
+  const typet &type,
+  const std::string &prefix="tmp");
+
+symbolt &c_declare_new_tmp_symbol(
+  code_blockt &init_code,
+  symbol_tablet &symbol_table,
+  const source_locationt &loc,
+  const typet &type,
+  const std::string &prefix="tmp");
```

Comment:
The above appear to be helper functions that ought to be just marked static and not be exported via the header file either.